Source/Core/DolphinWX/HideObjectAddEdit.cpp

```diff
@@ -0,0 +1,340 @@
+	wxButton* BruteForceDown = new wxButton(this, ID_BUTTON_DOWN, _("Down"));
+	wxString* BruteForceToolTip = new wxString(_("The up or down button can be used to find new codes.  While the game is playing, find an object you want to hide and select '8bits'.  Keep clicking the up button until the object disappears.  Now choose '16bits' and add two zeros to the right side of the 8bit code. "
+		"Continue clicking up until the object is hidden again.  Repeat this process until the code is long enough to be unique.  Warning: Too short of a code may result in other objects being hidden."));
+	BruteForceUp->SetToolTip(*BruteForceToolTip);
+	BruteForceDown->SetToolTip(*BruteForceToolTip);
+	BruteForceUp->Bind(wxEVT_BUTTON, &CHideObjectAddEdit::ButtonUporDown, this);
+	BruteForceDown->Bind(wxEVT_BUTTON, &CHideObjectAddEdit::ButtonUporDown, this);
+
+	wxBoxSizer* sEditHideObjectName = new wxBoxSizer(wxHORIZONTAL);
+	sEditHideObjectName->Add(EditHideObjectNameText, 0, wxALIGN_CENTER_VERTICAL|wxALL, 5);
+	sEditHideObjectName->Add(EditHideObjectName, 1, wxEXPAND|wxALL, 5);
+	sEditHideObject->Add(sEditHideObjectName, 0, wxEXPAND);
+	sbEntry = new wxStaticBoxSizer(wxVERTICAL, this, wxString::Format(_("Hide Objects Code")));
+
+	wxGridBagSizer* sgEntry = new wxGridBagSizer(0, 0);
+	sgEntry->Add(EditHideObjectType, wxGBPosition(0, 0), wxGBSpan(2, 5), wxEXPAND|wxALL, 5);
+	sgEntry->Add(EditHideObjectValueText, wxGBPosition(2, 0), wxGBSpan(1, 1), wxALIGN_CENTER_VERTICAL|wxALL, 5);
+	sgEntry->Add(EditHideObjectValue, wxGBPosition(2, 1), wxGBSpan(1, 1), wxEXPAND|wxALL, 5);
+	sgEntry->Add(BruteForceUp, wxGBPosition(2, 2), wxGBSpan(1, 1), wxEXPAND|wxALL, 5);
+	sgEntry->Add(BruteForceDown, wxGBPosition(2, 3), wxGBSpan(1, 1), wxEXPAND | wxALL, 5);
+	sgEntry->AddGrowableCol(1);
+
+	sbEntry->Add(sgEntry, 0, wxEXPAND);
+
+	sEditHideObject->Add(sbEntry, 0, wxEXPAND|wxALL, 5);
+	sEditHideObject->Add(CreateButtonSizer(wxOK | wxCANCEL), 0, wxEXPAND|wxLEFT|wxRIGHT|wxBOTTOM, 5);
+	SetSizerAndFit(sEditHideObject);
+	SetFocus();
+}
+
+void CHideObjectAddEdit::SaveHideObjectData(wxCommandEvent& event)
+{
+	if (!UpdateTempEntryData(itCurEntry))
+		return;
+
+	if (selection == -1)
+	{
+		for (int i = 0; i < HideObjectCodes.size(); ++i)
+		{
+			if (HideObjectCodes[i].name == WxStrToStr(EditHideObjectName->GetValue()))
+			{
+				wxMessageBox(_("Name is already in use.  Please choose a unique name."), _("Error"));
+				return;
+			}
+		}
+
+		HideObjectEngine::HideObject newHideObject;
+		newHideObject.name = WxStrToStr(EditHideObjectName->GetValue());
+		newHideObject.entries = tempEntries;
+		newHideObject.active = true;
+
+		HideObjectCodes.push_back(newHideObject);
+	}
+	else
+	{
+		HideObjectCodes[selection].name = WxStrToStr(EditHideObjectName->GetValue());
+		HideObjectCodes[selection].entries = tempEntries;
+	}
+
+	AcceptAndClose();
+	event.Skip();
+}
+
+void CHideObjectAddEdit::ButtonUporDown(wxCommandEvent& event)
+{
+	if (!UpdateTempEntryData(itCurEntry))
+		return;
+
```

Comment:
Repasting because it got removed in the rebase:

You may want to look into the use of validators, they would make all of this limit testing code much smaller. wxValidator or a subclass of it could be handy here.